pkg/lib/logging/logger.go

```diff
@@ -45,25 +46,81 @@ type Logger struct {
 	level int
 }
 
-// Get a named logger.
-func WithName(name string, kvpair ...interface{}) *Logger {
-	l := &Logger{
-		Real: Factory.New(),
-		name: name,
+type LevelLogger interface {
+	Info(msg string, kv ...interface{})
+	Enabled() bool
+	Error(err error, msg string, kv ...interface{})
+	WithValues(kv ...interface{}) LevelLogger
+	WithName(name string) LevelLogger
+	V(level int) LevelLogger
+	Trace(err error, kvpair ...interface{})
+}
+
+type levelLoggerImpl struct {
+	real  logr.Logger
+	level int
+}
+
+func (l *levelLoggerImpl) Info(msg string, kv ...interface{}) {
+	if Settings.allowed(l.level) {
+		l.real.Info(msg, kv...)
 	}
-	l.Real = l.Real.WithValues(kvpair...)
-	l.Real = l.Real.WithName(name)
+}
+
+func (l *levelLoggerImpl) Enabled() bool {
+	return l.real.Enabled()
+}
+
+func (l *levelLoggerImpl) Error(err error, msg string, kv ...interface{}) {
+	l.real.Error(err, msg, kv...)
+}
+
+func (l *levelLoggerImpl) V(level int) LevelLogger {
+	return &levelLoggerImpl{
+		real:  l.real.V(level),
+		level: level,
+	}
+}
+
+func (l *levelLoggerImpl) WithValues(kv ...interface{}) LevelLogger {
+	return &levelLoggerImpl{
+		real:  l.real.WithValues(kv...),
+		level: l.level,
+	}
+}
 
-	return l
+func (l *levelLoggerImpl) WithName(name string) LevelLogger {
+	return &levelLoggerImpl{
+		real:  l.real.WithName(name),
+		level: l.level,
+	}
+}
+
+func (l *levelLoggerImpl) Trace(err error, kvpair ...interface{}) {
+	l.real.Error(err, None, kvpair...)
```

Comment:
Trace messages are logged as Error? I'd expect the check of Settings.allowed to be here rather than on Info

Reply:
perhaps, but this is the current implementation, and not only that, it's used extensively to log errors 😅

Reply:
ok, if that's the current implementation.. where can I see that?

Reply:
I see, the comment for Trace is `// Logs an error without a description.` - so that's explain it.. anyway, if it keeps the existing behavior then it's fine